extension/intents/navigation/navigation.js

```diff
@@ -191,6 +191,38 @@ intentRunner.registerIntent({
   },
 });
 
+intentRunner.registerIntent({
+  name: "navigation.signInAndOut",
+  async run(context) {
+    const activeTab = await browserUtil.activeTab();
+    await content.inject(activeTab.id, [
+      "/intents/navigation/clickLogin.js",
+      "/intents/navigation/clickLogout.js",
+    ]);
+    if (context.parameters.action === "login") {
+      const found = await browser.tabs.sendMessage(activeTab.id, {
+        type: "signIn",
+        query: context.slots.query,
+      });
+      if (found === false) {
+        const exc = new Error("No link found matching query");
+        exc.displayMessage = `Sorry can't "${context.slots.query}" on this page`;
```

Comment:
There's no query slot on these phrases. You could use `context.utterance` here I believe. Please make sure to test on a page that has no sign in!

Reply:
Alright